lib/iris/fileformats/pp.py

```diff
@@ -1955,23 +1955,33 @@ def _convert_constraints(constraints):
     constraints = iris._constraints.list_of_constraints(constraints)
     pp_constraints = {}
     unhandled_constraints = False
+
+    def make_func(stashobj):
+        return lambda stash: stash==stashobj
+
     for con in constraints:
         if isinstance(con, iris.AttributeConstraint) and \
                 list(con._attributes.keys()) == ['STASH']:
             # Convert a STASH constraint.
+            # The attribute can be a STASH object, a stashcode string,or a 
+            # callable.
             stashobj = con._attributes['STASH']
-            if not isinstance(stashobj, STASH):
-                # The attribute can be a STASH object, or a stashcode string.
-                stashobj = STASH.from_msi(stashobj)
+            if callable(stashobj):
+                call_func = stashobj
+            else:
+                call_func = make_func(stashobj)
+
+            # Raise an exception if stashobj was not string or STASH object??   
```

Comment:
Wasn't sure whether to raise an exception here as the original code didn't and allowed any bad types to be picked up by the STASH.from_msi method. Advice on this would be much appreciated! :smile: